src/com/google/appengine/demos/shardedcounter/java/v2/ShardedCounter.java

```diff
@@ -243,7 +243,7 @@ private void incrementPropertyTx(final Key key, final String prop,
                     "You may need more shards. Consider adding more shards.");
             LOG.log(Level.WARNING, e.toString(), e);
             
-            if(getShardCount() <= ATMOST_SHARDS){
+            if(getShardCount() <= MAX_SHARDS){
                 LOG.log(Level.INFO,
                        "Doubling Shade, for this process");
                 addShards(getShardCount()); //double the shade by adding the number of shade again
```

Comment:
yes, but the MAX_SHARDS won't allow it to add more than MAX_SHARDS, so the shards won't accumulate with time when you encounter concurrent exception